waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -130,6 +139,361 @@ protected String getUsage() {
+
+            for (ServerDescriptor server : clusterManager.serverDescriptors()) {
+                CompletableFuture<Void> future = buildServerZKPartitionAssignmentsValidation(rpcClient, server, clusterManager,
+                         partitionValidationResultsList);
+                set.add(future);
+            }
+            return CompletableFuture.allOf(set.toArray(new CompletableFuture[set.size()]));
+        }
+
+        /**
+         * Status of the partition assignment for a server
+         */
+        enum AssignmentMatch {
+            ONLY_IN_ZOOKEEPER,
+            ONLY_IN_SERVER,
+            IN_BOTH
+        }
+
+        /**
+         * Compares waltz server partition assignment lists on Zookeeper metadata versus on actual servers
+         *
+         * @param zookeeperAssignments List of server partition assignment metadata on Zookeeper
+         * @param serverAssignments List of partition assignment on actual server
+         * @return Map containing the comparison result for each partition
+         */
+        private Map<Integer, AssignmentMatch> verifyAssignments(List<PartitionInfo> zookeeperAssignments,
+                                       List<Integer> serverAssignments) {
+
+            Map<Integer, AssignmentMatch> assignmentMatchHashMap = new HashMap<>();
+            for (PartitionInfo partitionInfo: zookeeperAssignments) {
+                assignmentMatchHashMap.put(partitionInfo.partitionId, AssignmentMatch.ONLY_IN_ZOOKEEPER);
+            }
+
+            for (Integer partitionId : serverAssignments) {
+                if (assignmentMatchHashMap.containsKey(partitionId)) {
+                    assignmentMatchHashMap.put(partitionId, AssignmentMatch.IN_BOTH);
+                } else {
+                    assignmentMatchHashMap.put(partitionId, AssignmentMatch.ONLY_IN_SERVER);
+                }
+            }
+            return assignmentMatchHashMap;
+        }
+
+        /**
+         * Parses the zookeeper servers partition assignments ZNode and insert the validation result into
+         * the partitionValidationResultsList for each partition
+         *
+         * @param zkClient Zookeeper client used to fetch Zookeeper partition assignment from
+         * @param zkRoot Zookeeper root path for cluster
+         * @param partitionValidationResultsList List indexed by partition in which validation outputs are inserted
+         * @throws ClusterManagerException Thrown if Zookeeper is missing some ZNodes or ZNode values
+         */
+        private void buildZookeeperPartitionAssignmentsValidation(ZooKeeperClient zkClient,
+                                                                  String zkRoot,
+                                                                  List<PartitionValidationResults> partitionValidationResultsList) throws ClusterManagerException {
+
+            ClusterManager clusterManager = new ClusterManagerImpl(zkClient, new ZNode(zkRoot), partitionAssignmentPolicy);
```

Comment:
why defining a new ClusterManager again?? In the processCmd() you have already defined ClusterManager.

Reply:
fixed. Thanks!